doc/tutorial/basic/index.md

```diff
@@ -246,108 +244,50 @@ Since we now have a method that is callable from JavaScript, all we have to do i
-usages of the DOM API with jQuery.
-
-### Depending on jQuery
-
-Just like for the DOM, there is a typed library for jQuery available in Scala.js: [scalajs-jquery](https://github.com/sjrd/scala-js-jquery) (there is [a livelier fork](https://github.com/exoego/scala-js-jquery) which you may prefer if it already supports the version of Scala you are using).
-Add the following line in your `build.sbt` by:
-
-{% highlight scala %}
-libraryDependencies += "be.doeraene" %%% "scalajs-jquery" % "0.9.6"
-{% endhighlight %}
-
-Don't forget to reload the sbt configuration now:
-
-1. Hit enter to abort the `~fastOptJS` command
-2. Type `reload`
-3. Start `~fastOptJS` again
-
-Again, make sure to update your IDE project files if you are using a plugin.
-
-### Using jQuery
-
-In `TutorialApp.scala`, remove the imports for the DOM, and add the import for jQuery:
-
-{% highlight scala %}
-import org.scalajs.jquery._
-{% endhighlight %}
-
-This allows you to easily access the `jQuery` main object of jQuery in your code (also known as `$`).
-
-We can now remove `appendPar` and replace all calls to it by the simple:
-
-{% highlight scala %}
-jQuery("body").append("<p>[message]</p>")
-{% endhighlight %}
-
-Where `[message]` is the string originally passed to `appendPar`, for example:
-
-{% highlight scala %}
-jQuery("body").append("<p>Hello World</p>")
-{% endhighlight %}
-
-If you try to reload your webpage now, it will not work (typically a `TypeError` would be reported in the console). The
-problem is that we haven't included the jQuery library itself, which is a plain JavaScript library.
-
-### Adding JavaScript libraries
-
-An option is to include `jquery.js` from an external source, such as [jsDelivr](http://www.jsdelivr.com/).
-
-{% highlight html %}
-<script type="text/javascript" src="http://cdn.jsdelivr.net/jquery/2.2.1/jquery.js"></script>
-{% endhighlight %}
+## <a name="setup-ui-in-scala-js"></a> Step 5: Setup the UI in Scala.js
 
-This can easily become very cumbersome, if you depend on multiple libraries. The Scala.js sbt plugin provides a
-mechanism for libraries to declare the plain JavaScript libraries they depend on and bundle them in a single file. All
-you have to do is activate this and then include the file.
+Previously, we have prepared the UI as an HTML document, then manipulated it from Scala.js code.
+That approach required us to manually *export* the method used for the `onclick` event of the button, which is cumbersome.
+We can avoid this issue by building the UI directly from the Scala.js code.
 
-In your `build.sbt`, set:
+Instead of prepare the button in the HTML, we can add the following code to the `main` method in Scala.js:
 
 {% highlight scala %}
-skip in packageJSDependencies := false
-jsDependencies +=
-  "org.webjars" % "jquery" % "2.2.1" / "jquery.js" minified "jquery.min.js"
+val button = document.createElement("button")
+button.textContent = "Click me!"
```

Comment:
In `appendPar` we seem to use `document.createTextNode` (which seems unnecessarily verbose). Is this inconsistency deliberate?

Reply:
No, it was not intended. The code in `appendPar` is much older, and perhaps at the time `textContent` did not exist yet. I have changed `appendPar` to use `textContent` a well.